File: STIF_Algo_bk/HMM_package/pgms/zscore_all_files.pl
#!/usr/bin/perl

$perl_script = $ARGV[0];

opendir DIR,"$ARGV[1]";

@dir = <DIR>;

while ($i = readdir(DIR))
{
        if($i =~ /(\w+.?\w+)_valid.outout\d*/)

        {
              print "SS :: $i :: $1\n";
              $file_name = $1;
              $cmd1 = "perl $perl_script\/zscore.pl $ARGV[1]\/$i $ARGV[1] \> $ARGV[1]\/$file_name\.zsco";
              # print "$cmd1\n";
              system($cmd1);

              $n++;
        }
}
print "CNT :: $n\n";
closedir(DIR);

File: STIF_Algo/HMM_package/pgms/zscore_all_files.py
#!/usr/bin/env python3

import sys
import os
import re

perl_script = sys.argv[1]
directory = sys.argv[2]

try:
    with os.scandir(directory) as dir_entries:
        n = 0
        for entry in dir_entries:
            if entry.is_file() and re.match(r'(\w+.?\w+)_valid.outout\d*', entry.name):
                file_name = entry.name
                base_name = re.match(r'(\w+.?\w+)_valid.outout\d*', file_name).group(1)
                print(f"SS :: {file_name} :: {base_name}")
                cmd1 = f"perl {perl_script}/zscore.pl {directory}/{file_name} {directory} > {directory}/{base_name}.zsco"
                # print(cmd1)
                os.system(cmd1)
                n += 1
        print(f"CNT :: {n}")
except FileNotFoundError:
    print(f"Error: Directory {directory} not found")
    sys.exit(1)
